src/Workspaces/Remote/ServiceHub/Services/AssetSynchronization/RemoteAssetSynchronizationService.cs

```diff
@@ -12,8 +12,14 @@
 
 namespace Microsoft.CodeAnalysis.Remote
 {
-    internal class RemoteAssetSynchronizationService : BrokeredServiceBase, IRemoteAssetSynchronizationService
+    internal sealed class RemoteAssetSynchronizationService : BrokeredServiceBase, IRemoteAssetSynchronizationService
     {
+        internal sealed class Factory : FactoryBase<IRemoteAssetSynchronizationService>
```

Comment:
📝 Nested types at the bottom or in a separate file.

Reply:
Such style is not followed in the code base. Specifically, in this case it does not make sense either since the Factory and the constructor should be together as they are closely related. I don't want to be jumping from bottom to the top of the file in between them.